lib/dispatcher/client.js

```diff
@@ -68,6 +69,10 @@ const getDefaultNodeMaxHeaderSize = http &&
 const noop = () => {}
 
 function getPipelining (client) {
+  if (client[kHTTPContext]?.version === 'h2' && client[kMaxConcurrentStreamsManual]) {
```

Comment:
Let's just add a deprecation warning if `maxConcurrentStreams` is not set, and we should be good to go

Reply:
When would that happen? The symbol (which name I don't like yet :see_no_evil: ) that I'm using is only set if `maxConcurrentStreams` is set (line 240).

Or your suggestion is to notify everyone using `allowH2` but not setting it - therefore missing the benefit of multiplexing?

I'm also not sure WHEN that deprecation warning would happen. I assume that somewhere in Client constructor. Otherwise the noise would be significant.

Reply:
I was thinking to add it only when the `kMaxConcurrentStreamsManual` is set to `false` but `pipeline` was altered instead; so the warning states that in further major `pipeline` will not take the effect of `maxConcurrentStreams`